src/content-helper/editor-sidebar/editor-sidebar.tsx

```diff
@@ -24,10 +30,106 @@ import { RelatedTopPostList } from './related-top-posts/component-list';
 
 const BLOCK_PLUGIN_ID = 'wp-parsely-block-editor-sidebar';
 
+/**
+ * Defines the data structure exposed by the Sidebar about the currently opened
+ * Post.
+ *
+ * @since 3.11.0
+ */
+export interface SidebarPostData {
+	authors: string[];
+	categories: string[];
+	tags: string[];
+}
+
+/**
+ * Defines typings for some non-exported Gutenberg functions to avoid
+ * intellisense errors in function calls.
+ *
+ * This can be removed once Gutenberg provides typings for these functions.
+ *
+ * @since 3.11.0
+ */
+interface GutenbergFunction {
+	getCurrentPostId(): number;
+	getEditedPostAttribute( attribute: string ): number[];
+}
+
+/**
+ * Returns the Content Helper Editor Sidebar.
+ *
+ * @since 3.4.0
+ *
+ * @return {JSX.Element} The Content Helper Editor Sidebar.
+ */
 const ContentHelperEditorSidebar = (): JSX.Element => {
 	const [ period, setPeriod ] = useState<Period>( Period.Days7 );
 	const [ metric, setMetric ] = useState<Metric>( Metric.Views );
+	const [ postData, setPostData ] = useState<SidebarPostData>( {
+		authors: [], categories: [], tags: [],
+	} );
+
+	/**
+	 * Returns the current Post's ID, tags and categories.
+	 *
+	 * @since 3.11.0
+	 */
+	const { postId, tags, categories } = useSelect( ( select ) => {
+		const { getCurrentPostId, getEditedPostAttribute } = select( editorStore ) as GutenbergFunction;
+		const { getEntityRecords } = select( coreStore );
+
+		const tagRecords: Taxonomy[]|null = getEntityRecords(
+			'taxonomy', 'post_tag', { include: getEditedPostAttribute( 'tags' ) }
+		);
+
+		const categoryRecords: Taxonomy[] | null = getEntityRecords(
+			'taxonomy', 'category', { include: getEditedPostAttribute( 'categories' ) }
+		);
+
+		return {
+			postId: getCurrentPostId(),
+			tags: tagRecords,
+			categories: categoryRecords,
+		};
+	}, [] );
+
+	/**
+	 * Returns the current Post's tag names.
+	 *
+	 * @since 3.11.0
+	 */
+	const tagNames = useMemo( () => {
+		return tags ? tags.map( ( t ) => t.name ) : [];
+	}, [ tags ] );
+
+	/**
+	 * Returns the current Post's category names.
+	 *
+	 * @since 3.11.0
+	 */
+	const categoryNames = useMemo( () => {
+		return categories ? categories.map( ( c ) => c.name ) : [];
+	}, [ categories ] );
+
+	useEffect( () => {
+		apiFetch<SidebarPostData>( {
+			path: addQueryArgs( '/wp-parsely/v1/post-data/', { postId } ),
+		} ).then( ( response ) => {
+			setPostData( {
+				authors: response.authors,
+				tags: tagNames,
+				categories: categoryNames,
+			} );
+		} );
+	}, [ postId, tagNames, categoryNames ] );
```

Comment:
Do we need to have these dependencies? If I'm understanding correctly, it will do a API fetch each time there's a change in tags or categories, and I'm not sure if it's necessary, since we already have those editor store.

Reply:
Here's what I've come to understand as this is new to me: When `useEffect` runs, any of these dependencies can be initially null as they are being resolved in the `useSelect` block. Then, they will resolve and they might do so in different order. So the array here is needed to get the final postData correctly as different parts of the request resolve.

Reply:
Ah makes sense! Thanks for the explanation!